### PR TITLE
[Backport] Admin user auth controller refactor

### PR DESCRIPTION
**Pull request by gelanivishal**
### Original Pull Request
https://github.com/magento/magento2/pull/16560
### Description
Refactor the code by removing the direct use of ObjectManager and includes the dependencies using Constructor Dependency Injection

### Manual testing scenarios
1. Created a new admin user
2. Login and logout using new admin user
3. Clicked on forgot password and provided new admin user email id.
4. Clicked on reset password link in email.
5. Created new password from reset password page.
6. Login using new password.

### CODE REVIEW

File: app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php [hunk 1]
```diff
@@ -7,26 +7,54 @@
...
+    /**
+     * @var Data
+     */
+    private $backendDataHelper;
+
+    /**
+     * @param Context $context
+     * @param UserFactory $userFactory
+     * @param SecurityManager $securityManager
+     * @param CollectionFactory $userCollectionFactory
```
Comment (sidolov) [inline app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php]: Please, add missing parameter to doc block

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php (hunk 1 above)
Comment (gelanivishal) [reply app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php]: Done.

**Review by gelanivishal — COMMENTED** [review]
@sidolov I have added missing parameter, Please review it.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-2557](https://jira.corp.magento.com/browse/ENGCOM-2557) has been created to process this Pull Request

**Comment by sidolov** [comment]
Hi @gelanivishal , one of static tests failing with provided changes: 
```
Magento\Test\Integrity\Di\CompilerTest::testConstructorIntegrity
Passed: 7792, Failed: 2, Incomplete: 0, Skipped: 0.
Data set: array (
  0 => '\\Magento\\User\\Controller\\Adminhtml\\Auth\\Forgotpassword',
)
Incorrect dependency in class \Magento\User\Controller\Adminhtml\Auth\Forgotpassword in /var/www/html/app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php
\Magento\Backend\Helper\Data already exists in context object

Data set: array (
  0 => '\\Magento\\User\\Controller\\Adminhtml\\Auth\\ResetPasswordPost',
)
Incorrect dependency in class \Magento\User\Controller\Adminhtml\Auth\ResetPasswordPost in /var/www/html/app/code/Magento/User/Controller/Adminhtml/Auth/ResetPasswordPost.php
\Magento\Backend\Helper\Data already exists in context object
```

Please, take a look and fix your solution according to test rules.
Thank you!

**Comment by sidolov** [comment]
Hi @gelanivishal , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**